kore/src/Kore/Step/Step.hs

```diff
@@ -224,8 +229,26 @@ unifyRule
     let
         RulePattern { requires = ruleRequires } = rule'
         requires' = Predicate.fromPredicate ruleRequires
-    unification' <- Substitution.normalizeExcept (unification <> requires')
+    unification' <- removeRedundancy unification requires'
     return (rule' `Conditional.withCondition` unification')
+  where
+    removeRedundancy unificationSolution requires = do
+        let predicate =
+                (makePredicate `on` Predicate.toPredicate)
+                    unificationSolution requires
+        isPreRedundant <- evaluate predicate
+        let nonRedundantPredicate =
+                case isPreRedundant of
+                    Just True -> unificationSolution
```

Comment:
So, I'm a bit confused by this. I think we only implemented the "Just False" case, see https://github.com/kframework/kore/blob/master/kore/src/Kore/Step/SMT/Evaluator.hs#L143
so I would expect this to be a no-op. On the other hand, your change seems to make a difference for the tests below.

So, I'm not sure what comment to make here, but this does not seem right.